### PR TITLE
Material_Mixed refactor

### PR DESCRIPTION
**Pull request by stevk**

### CODE REVIEW

File: Source/ModelGroup.cs [hunk 1]
```diff
@@ -17,18 +17,26 @@ internal abstract class ModelGroup
         public int Id = -1;
         public bool NoSampleImages = false;
 
-        protected Runtime.Image GetImage(List<string> imageList, string name)
+        protected Runtime.Image GetImage(List<string> imageList, string name, bool isFigure = false)
```
Comment (bghgary) [inline Source/ModelGroup.cs]: rename to UseImage

File: Source/ModelGroup.cs [hunk 2]
```diff
@@ -71,7 +80,59 @@ public static Runtime.MeshPrimitive CreateSinglePlane()
                     {
                         1, 0, 3, 1, 3, 2
                     },
-            };
+                };
+            }
+
+            public static List<Runtime.MeshPrimitive> CreatePlaneFromTriangles()
```
Comment (bghgary) [inline Source/ModelGroup.cs]: rename to CreateMultiPrimitivePlane

File: Source/ModelGroup.cs (hunk 1 above)
Comment (bghgary) [reply Source/ModelGroup.cs]: also make this private and create two functions UseTexture and UseFigure.

File: Source/ModelGroups/Material_Mixed.cs [hunk 3]
```diff
@@ -1,236 +1,103 @@
...
-                    if (wrapper.ExtensionsUsed == null)
-                    {
-                        wrapper.ExtensionsUsed = new List<string>();
-                    }
-                    wrapper.ExtensionsUsed = wrapper.ExtensionsUsed.Union(
-                        new string[] { "KHR_materials_pbrSpecularGlossiness" }).ToList();
-                }
-                else if (req.name == Propertyname.BaseColorTexture)
+                // Apply the common properties to the gltf.
+                meshPrimitives[0].Material.MetallicRoughnessMaterial.BaseColorTexture = new Runtime.Texture { Source = baseColorTextureImage };
```
Comment (bghgary) [inline Source/ModelGroups/Material_Mixed.cs]: create new Texture and set both to the same variable

File: Source/ModelGroups/Material_Mixed.cs [hunk 4]
```diff
@@ -1,236 +1,103 @@
...
-                }
+                            new Runtime.Node
+                            {
+                                Mesh = new Runtime.Mesh
+                                {
+                                    MeshPrimitives = meshPrimitives
+                                },
+                            },
+                        },
+                    },
```
Comment (bghgary) [inline Source/ModelGroups/Material_Mixed.cs]: }, new[] { "KHR_materials_pbrSpecularGlossiness" }),

File: Source/ModelGroups/Material_Mixed.cs [hunk 5]
```diff
@@ -1,236 +1,103 @@
...
-                else if (property.propertyGroup == 0
-                    )
-                {
-                    wrapper.Scenes[0].Nodes[0].Mesh.MeshPrimitives = new List<Runtime.MeshPrimitive>
-                    {
-                        prim0,
-                        prim1
-                    };
-                }
+                material0.Extensions = new List<Runtime.Extensions.Extension>() { new Runtime.Extensions.PbrSpecularGlossiness() }; ;
```
Comment (bghgary) [inline Source/ModelGroups/Material_Mixed.cs]: double semicolon all over

File: Source/ModelGroup.cs (hunk 1 above)
Comment (stevk) [reply Source/ModelGroup.cs]: Fixed

File: Source/ModelGroup.cs (hunk 2 above)
Comment (stevk) [reply Source/ModelGroup.cs]: Fixed

File: Source/ModelGroups/Material_Mixed.cs (hunk 3 above)
Comment (stevk) [reply Source/ModelGroups/Material_Mixed.cs]: fixed

File: Source/ModelGroups/Material_Mixed.cs (hunk 4 above)
Comment (stevk) [reply Source/ModelGroups/Material_Mixed.cs]: Fixed

File: Source/ModelGroups/Material_Mixed.cs (hunk 5 above)
Comment (stevk) [reply Source/ModelGroups/Material_Mixed.cs]: Fixed

**Pull request merged**